kernel/src/platform/mod.rs

```diff
@@ -60,6 +60,43 @@ pub trait Platform {
     ) -> Result<(), returncode::ReturnCode> {
         Ok(())
     }
+
+    /// This function is called when an app faults
+    ///
+    /// This is an optional function that can be implemented by `Platform`s that
+    /// allows the chip to handle the app fault and not terminate or restart
+    /// the app.
+    ///
+    /// If `Ok(())` is returned by this function then the kernel will not
+    /// terminate or restart the app, but instead allow it to continue
+    /// running. NOTE in this case the chip must have fixed the underlying
+    /// reason for fault otherwise it will re-occur.
```

Comment:
I think we need to update the comments for `ContextSwitchReason` to clarify that when returning the fault case the UKB implementation should leave things in a state where the kernel can re-run the last instruction.